Sources/SwiftKueryPostgreSQL/PostgreSQLConnection.swift

```diff
@@ -378,38 +412,50 @@ public class PostgreSQLConnection: Connection {
         }
         else if let preparedStatement = preparedStatement {
             guard let statement = preparedStatement as? PostgreSQLPreparedStatement else {
-                onCompletion(.error(QueryError.unsupported("Failed to execute unsupported prepared statement")))
+                self.runCompletionHandler(.error(QueryError.unsupported("Failed to execute unsupported prepared statement")), onCompletion: onCompletion)
                 return
             }
-            
-            if !preparedStatements.contains(statement.name) {
-                if let error = prepareStatement(name: statement.name, for: statement.query) {
-                    onCompletion(.error(QueryError.databaseError(error)))
-                    return
 
+            guard preparedStatements.contains(statement.name) else {
+                prepareStatement(statement.name, statement.query) { result in
+                    if let error = result.asError {
+                        return self.runCompletionHandler(.error(QueryError.databaseError(error.localizedDescription)), onCompletion: onCompletion)
+                    }
+                    _ = parameterData.withUnsafeBufferPointer { buffer in
```

Comment:
Rather than duplicating the next 9 lines of code within the `prepareStatement` callback, could this callback just call `execute` again, now that we know the statement is cached on this connection?

To do this, you'd need to refactor this method so that the check for preparing a statement is the first thing done (before parameter pointers are allocated).

Because we are going to change this function itself to offload (later, after the check for preparing a statement), we will end up offloading twice in the case where the statement is not yet prepared on the current connection.  However, this will only occur once per connection per statement, so I think this is okay.